subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_aux.c

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (c) 2018-2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stddef.h>
+
+#include <toolchain.h>
+#include <zephyr/types.h>
+#include <sys/byteorder.h>
+
+#include "hal/ccm.h"
+#include "hal/radio.h"
+#include "hal/ticker.h"
+
+#include "util/util.h"
+#include "util/memq.h"
+
+#include "pdu.h"
+
+#include "lll.h"
+#include "lll_vendor.h"
+#include "lll_clock.h"
+#include "lll_chan.h"
+#include "lll_adv.h"
+#include "lll_adv_aux.h"
+
+#include "lll_internal.h"
+#include "lll_adv_internal.h"
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HCI_DRIVER)
+#define LOG_MODULE_NAME bt_ctlr_lll_adv_aux
+#include "common/log.h"
+#include <soc.h>
+#include "hal/debug.h"
+
+static int init_reset(void);
+static int prepare_cb(struct lll_prepare_param *prepare_param);
+static void abort_cb(struct lll_prepare_param *prepare_param, void *param);
+
+int lll_adv_aux_init(void)
+{
+	int err;
+
+	err = init_reset();
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+int lll_adv_aux_reset(void)
+{
+	int err;
+
+	err = init_reset();
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+void lll_adv_aux_prepare(void *param)
+{
+	struct lll_prepare_param *p = param;
+	int err;
+
+	err = lll_hfclock_on();
+	LL_ASSERT(!err || err == -EINPROGRESS);
+
+	err = lll_prepare(lll_is_abort_cb, abort_cb, prepare_cb, 0, p);
+	LL_ASSERT(!err || err == -EINPROGRESS);
+}
+
+void lll_adv_aux_offset_fill(u32_t ticks_offset, u32_t start_us,
```

Comment:
Could this function be moved to ull_adv_ext to reduce code copying? And if so, could it return the aux pointer for convenience?

Reply:
yes, moved the function to ULL files.